src/jit/morph.cpp

```diff
@@ -8632,115 +8637,20 @@ GenTreePtr Compiler::fgMorphCall(GenTreeCall* call)
             // Peel off casts
             while (treeWithCall->gtOper == GT_CAST)
             {
-                noway_assert(!treeWithCall->gtOverflow());
+                assert(!treeWithCall->gtOverflow());
                 treeWithCall = treeWithCall->gtGetOp1();
             }
 
-            noway_assert(treeWithCall == call);
+            assert(treeWithCall == call);
```

Comment:
I think maybe these should stay as `noway_asserts` ?

Reply:
It is `ifdef DEBUG`, so I think we do not want to use `noway_assert` in debug-only code.

Reply:
Ah, good point.